cpp/src/arrow/dataset/scanner.h

```diff
@@ -37,6 +38,7 @@ namespace dataset {
 /// \brief Shared state for a Scan operation
 struct ARROW_DS_EXPORT ScanContext {
   MemoryPool* pool = arrow::default_memory_pool();
+  internal::ThreadPool* thread_pool = arrow::internal::GetCpuThreadPool();
```

Comment:
Maybe this should be a task group instead of a thread pool. Then users can pass a serial task group to signal single threaded operation